### PR TITLE
Build with static rainbow

### PR DESCRIPTION
**Pull request by oynqr**
This should let rainbow build statically even on Windows.
I have a separate branch that moves rainbow in-tree if that is better.

### CODE REVIEW

**Comment by Scrumplex** [comment]
See review beloww

File: libraries/rainbow/CMakeLists.txt [hunk 1]
```diff
@@ -8,15 +8,7 @@ set(RAINBOW_SOURCES
 src/rainbow.cpp
 )
 
-add_definitions(-DRAINBOW_LIBRARY)
```
Comment (Scrumplex) [inline libraries/rainbow/CMakeLists.txt:11]: Maybe revert 61db1c46beb465c33124ec4f34dfdcefd4d804d3 first and then apply your changes. That way all libraries should be called `Launcher_`. 61db1c46beb465c33124ec4f34dfdcefd4d804d3 was just a workaround to allow installing PolyMC and MMC side-by-side

**Review by Scrumplex — CHANGES_REQUESTED** [review]

**Review by Scrumplex — APPROVED** [review]

**Comment by DioEgizio** [comment]
huh appimage is failing

>ERROR: Could not find dependency: libLauncher_rainbow.so

File: libraries/rainbow/CMakeLists.txt [hunk 2]
```diff
@@ -8,15 +8,7 @@ set(RAINBOW_SOURCES
...
-add_library(PolyMC_rainbow SHARED ${RAINBOW_SOURCES})
-target_include_directories(PolyMC_rainbow PUBLIC "${CMAKE_CURRENT_SOURCE_DIR}/include")
+add_library(Launcher_rainbow SHARED ${RAINBOW_SOURCES})
```
Comment (Scrumplex) [inline libraries/rainbow/CMakeLists.txt]:
```suggestion
add_library(Launcher_rainbow STATIC ${RAINBOW_SOURCES})
```
That should fix the issue

**Review by Scrumplex — CHANGES_REQUESTED** [review]

**Comment by oynqr** [comment]
God damn it, git pulled a dirty one on me 😂

**Comment by DioEgizio** [comment]
Looks like it had the opposite effect lol

**Comment by Scrumplex** [comment]
The actions are extremely sad

File: launcher/CMakeLists.txt [hunk 3]
```diff
@@ -941,7 +941,6 @@ target_link_libraries(Launcher_logic
     Launcher_iconfix
     QuaZip::QuaZip
     hoedown
-    PolyMC_rainbow
```
Comment (Scrumplex) [inline launcher/CMakeLists.txt:944]: uhhh

**Review by Scrumplex — CHANGES_REQUESTED** [review]

File: launcher/CMakeLists.txt (hunk 3 above)
Comment (oynqr) [reply launcher/CMakeLists.txt:944]: What the fuck is even going on, I removed that

God, I need a break

Comment (Scrumplex) [reply launcher/CMakeLists.txt:944]: It should be `Launcher_rainbow`. Your second to last force push removed it for some reason

**Comment by Scrumplex** [comment]
Let's hope it's fine now :D

**Comment by oynqr** [comment]
It should be, shoutout goes to git merge for being garbo and also to my monitor that won't display vimdiff in a legible manner.

**Review by Scrumplex — APPROVED** [review]
works on my machine :tm:

**Review by d-513 — APPROVED** [review]

**Review by txtsd — APPROVED** [review]
Good job getting this to work!

**Pull request merged**